projects/com.oracle.truffle.llvm.asm.amd64/src/com/oracle/truffle/llvm/asm/amd64/AsmFactory.java

```diff
@@ -156,6 +158,26 @@ public void createUnaryOperation(String operation, String node) {
         this.statements.add(LLVMWriteI32NodeGen.create(opNode, slot));
     }
 
+    public void createDivisionOperatoin(String op, String divisor) {
+        FrameSlot slot = frameDescriptor.findFrameSlot(divisor);
+        LLVMI32Node divisorNode = (slot != null) ? LLVMI32ReadNodeGen.create(slot) : getImmediateNode(divisor);
+        LLVMNode statement = new LLVMI32UnsupportedInlineAssemblerNode();
```

Comment:
I think it would be easier to read if you put creating the `LLVMI32UnsupportedInlineAssemblerNode` in an else block after `op.equals("idivl")`. Otherwise, you first create a node that gets overwritten again when `op.equals("idivl")`. If you want, you can address this with the next PR.